chargebee/src/main/java/com/chargebee/android/resources/ReceiptResource.kt

```diff
@@ -12,11 +12,13 @@ import com.chargebee.android.responseFromServer
 internal class ReceiptResource : BaseResource(baseUrl = Chargebee.baseUrl){
 
     internal suspend fun validateReceipt(params: Params): ChargebeeResult<Any> {
-        var dataMap = mapOf<String, String>()
+        var dataMap = mapOf<String, String?>()
         val paramDetail = CBReceiptRequestBody.fromCBReceiptReqBody(params)
         dataMap = if (!(TextUtils.isEmpty(params.customerId))) {
             paramDetail.toCBReceiptReqBody()
-        }else{
+        } else if (params.customer != null){
+            paramDetail.toCBReceiptReqCustomerBody()
+        } else{
```

Comment:
We could refactor this as well if we remove the customerId property.

Reply:
yes, will remove it once we remove the old `purchaseProduct()` method.